### ISSUE TITLE
ng build --prod fails when HttpInterceptorModule imported in app.module

### ISSUE THREAD

**Issue by amrendr** [gund/ng-http-interceptor#152]
Just by adding the following statement on my app.module file, production build with AOT fails.

``` ts
import { HttpInterceptorModule } from 'ng-http-interceptor';
```
running production  build, I am getting following error. 

```ts
ERROR in D:/Projects/Git-Repos/ui/src/$$_gendir/node_modules/
ng-http-interceptor/dist/ng-http-interceptor.ngfactory.ts (10,26): 
Cannot find module 'httpInterceptor'.
```

I have following in my dependencies
"ng-http-interceptor": "^3.1.0",
 "proxy-polyfill": "^0.1.7",

and following in devdependencies
"tslib": "1.6.0",

Can you please help me know, what may I doing wrong?

**Comment by imgx64** [comment]
This seems to be a mismatch between the npm package's name and some build option (either [this](https://github.com/gund/ng-http-interceptor/blob/a5dcbca9f666a301e7e7495b19d2890f7726c5ec/rollup.config.js#L8) or [this](https://github.com/gund/ng-http-interceptor/blob/a5dcbca9f666a301e7e7495b19d2890f7726c5ec/tsconfig.es2015.json#L25), I haven't investigated further). 

If I manually rename the folder `./node-modules/ng-http-interceptor` to `httpInterceptor`, AOT works correctly.

**Comment by gund** [comment]
Hmm, weird...
I might have an idea why it can happen.
When ngc generates aot metadata it asks for the name of library, so I set it to be httpInterceptor.
Maybe it should match the directory name...

**Comment by alperg** [comment]
I was having this issue with my Jenkins build setup (no issues locally) and fixed by adding the following shell commands to my post-build process as @imgx64 suggested:

```bash
mkdir ./node_modules/httpInterceptor
cp -r ./node_modules/ng-http-interceptor/* ./node_modules/httpInterceptor 
```

**Comment by gund** [comment]
@alperg I guess on your local machine you have version below v3.1.x so AOT there works fine.
It appears to be broken on v3.1.0 when I updated the way I generate AOT artifacts...

For the time being you can just stick with version v3.0.x until I will publish a fixed version.

**Comment by gund** [comment]
Okay so I already found a problem. Will share here the details. Meanwhile fix will come soon.

So in TS config file string `angularCompilerOptions -> flatModuleId` is used to set the `importAs` property in top level `*.metadata.json` file which dictates to NGC where to look for the library imports.

So essentially I was treating this property wrong, I was thinking about it as the name of library when it is used in module loaders like AMD (like rollup's moduleName property in config).
But it should match exactly the name of the import apparently =)

**Comment by gund** [comment]
Fix landed in [v3.1.1](https://github.com/gund/ng-http-interceptor/releases/tag/v3.1.1)
